test/unchecked.jl

```diff
@@ -0,0 +1,155 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# unchecked integer arithmetic
+
+import Base.Unchecked:
+    unchecked_abs, unchecked_neg, unchecked_add, unchecked_sub,
+    unchecked_mul, unchecked_div, unchecked_rem, unchecked_fld,
+    unchecked_mod, unchecked_cld
+
+# Test return types
+for T in (Int8,Int16,Int32,Int64,Int128, UInt8,UInt16,UInt32,UInt64,UInt128)
+    z, o = T(0), T(1)
+    @test typeof(unchecked_neg(z)) === T
```

Comment:
I use `===` for types since I have a hunch it's a faster (since it is a simpler notion of equality).

I do need it when comparing integers since I also want to check whether the types are what I expect, e.g. that `Int8 + Int8 == Int8`, so to say.